### PR TITLE
clean pytorch backend.

### PR DESCRIPTION
**Pull request by SaitejaUtpala**
Pytorch 1.9 has just been released. This PR upgrades torch version to 1.8.1 and cleans pytorch background.

### CODE REVIEW

File: geomstats/_backend/pytorch/__init__.py [hunk 1]
```diff
@@ -28,13 +31,15 @@
...
-    gt as greater,
+    outer,
+    greater,
+    hstack,
     int32,
     int64,
     isnan,
     log,
-    lt as less,
+    logical_or,
```
Comment (nguigs) [inline geomstats/_backend/pytorch/__init__.py:41]: `torch.logical_and` seems to exist as well: https://pytorch.org/docs/1.8.1/generated/torch.logical_and.html?highlight=logical_and#torch.logical_and

File: geomstats/tests.py [hunk 2]
```diff
@@ -68,14 +68,29 @@ def np_and_pytorch_only(test_item):
     import tensorflow as tf
     _TestBaseClass = tf.test.TestCase
 
+if pytorch_backend():
+    import torch
+
+def pytorch_error_msg(a, b):
```
Comment (nguigs) [inline geomstats/tests.py:74]: Good idea!

**Review by nguigs — APPROVED** [review]
Great!

File: geomstats/_backend/pytorch/__init__.py (hunk 1 above)
Comment (SaitejaUtpala) [reply geomstats/_backend/pytorch/__init__.py:41]: > `torch.logical_and` seems to exist as well: https://pytorch.org/docs/1.8.1/generated/torch.logical_and.html?highlight=logical_and#torch.logical_and

@nguigs yup. But somehwere in the codebase, `logical_and` is used for non-torch tensor type and so we have to keep what is currently. 

```
def logical_and(x, y):
    if torch.is_tensor(x):
        return x & y
    return x and y

```

Comment (SaitejaUtpala) [reply geomstats/_backend/pytorch/__init__.py:41]: Actually. Quite a bit of code can be removed if we don't use these backend related functions in heterogeneous way  (mixing numpy arrays, torch tensors, lists and passing to backend functions)

File: geomstats/tests.py (hunk 2 above)
Comment (ninamiolane) [reply geomstats/tests.py:74]: Yay!! 🎉

File: opt-requirements.txt [hunk 3]
```diff
@@ -1,3 +1,3 @@
 tensorflow>=2.1.2; python_version < '3.8'
+torch==1.8.1
```
Comment (ninamiolane) [inline opt-requirements.txt:2]: import order

File: geomstats/_backend/pytorch/__init__.py [hunk 4]
```diff
@@ -155,13 +161,9 @@ def split(x, indices_or_sections, axis=0):
     return torch.split(x, tuple(intervals_length), dim=axis)
 
 
-def logical_or(x, y):
-    return x or y
-
-
 def logical_and(x, y):
```
Comment (ninamiolane) [inline geomstats/_backend/pytorch/__init__.py:164]: what is happening with codecov? the coverage is shown to decrease on this PR, and for example it seems that the __init__.py file has coverage 0%... :/

**Review by ninamiolane — APPROVED** [review]
Awesome work, thank you @SaitejaUtpala !! 🎉 

I'm just confused about the codecoverage: do you know what is happening there?

**Comment by SaitejaUtpala** [comment]
@ninamiolane  RTM.

**Pull request merged**